programs/ssl/ssl_server2.c

```diff
@@ -220,9 +226,38 @@ int main( void )
 #endif /* MBEDTLS_SSL_ASYNC_PRIVATE */
 
 #if defined(MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED)
-#define USAGE_PSK                                                   \
+#define USAGE_PSK_RAW                                               \
     "    psk=%%s              default: \"\" (in hex, without 0x)\n" \
-    "    psk_identity=%%s     default: \"Client_identity\"\n"
+    "    psk_identity=%%s     default: \"Client_identity\"\n"       \
+    "    psk_list=%%s         default: \"\"\n"                          \
+    "                          A list of (PSK identity, PSK value) pairs in (hex format, without 0x)\n" \
+    "                          id1,psk1[,id2,psk2[,...]]\n"
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+#define USAGE_PSK_SLOT                          \
+    "    psk_slot=%%d         default: 0 (don't use key slots)\n"     \
+    "                          An empty key slot identifier to be used to hold the static PSK\n" \
+    "                          configured through the psk parameter.\n"\
+    "                          Note: Currently only supported in conjunction with\n"                  \
+    "                          the use of min_version to force TLS 1.2 and force_ciphersuite \n"      \
+    "                          to force a particular PSK-only ciphersuite.\n"                         \
+    "                          Note: This is to test integration of PSA-based opaque PSKs with\n"     \
+    "                          Mbed TLS only. Production systems are likely to configure Mbed TLS\n"  \
+    "                          with prepopulated key slots instead of importing raw key material.\n" \
+    "    psk_list_slot=%%d    default: 0 (don't use key slots)\n"     \
```

Comment:
Please consider removing this option and selecting slots dynamically instead.

Reply:
I don't want to remove this option to allow us testing opaque static PSKs but raw dynamic ones, and vice versa. But as for `psk_slot` I will turn it into a boolean option, if you're ok with that.

Reply:
Oh, good point about testing a mix. Please keep that option then, but just rename it and turn it to a boolean.